### PR TITLE
Ticket Cancellation

### PR DESCRIPTION
**Pull request by gem-aditya-Saraswat**

### CODE REVIEW

File: src/main/java/selectors/Calender_Xpath.java
Comment (gem-akhil-singh) [inline src/main/java/selectors/Calender_Xpath.java:1]: Change Class Name, follow Pascal Case and Use CamelCase in Selector Variable Names

File: src/main/java/selectors/Sorting_Xpath.java
Comment (gem-akhil-singh) [inline src/main/java/selectors/Sorting_Xpath.java:1]: Change ClassName Use Pascal Case

File: src/main/java/stepdefination/Calender_StepDefination.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED", STATUS.FAIL);
+        }
+    }
+
+    @Then("^User selected date range with (.+) and (.+)$")
+    public void dateRangeSelector(String startDate, String endDate){
+        try{
+            Calender_Utils calUtils = new Calender_Utils();
+            calUtils.select_specified_date(startDate);
+            calUtils.select_specified_date(endDate);
```
Comment (gem-akhil-singh) [inline src/main/java/stepdefination/Calender_StepDefination.java:170]: Here , Gem Test Reporter not used for Reporting , It was needed or not ?

File: src/main/java/stepdefination/Login_StepDefination.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.slf4j.LoggerFactory;
+import io.cucumber.java.en.Given;
+import static com.gemini.generic.ui.utils.DriverAction.*;
+import io.cucumber.java.en.Then;
+import selectors.Login_Xpath;
+import utilities.Login_Utils;
+import utilities.Read_Properties;
+
+
+public class Login_StepDefination extends GemjarTestngBase {
```
Comment (gem-akhil-singh) [inline src/main/java/stepdefination/Login_StepDefination.java:16]: Speliing of Definition and use pascal case in the Class name

**Review by gem-akhil-singh — CHANGES_REQUESTED** [review]
* Do follow the mentioned comments 
* Use Pascal Case in classnames , and camelCase in the variable names there are lot of files which are left 
* You test Code should be inside src/test not src/main
* Please dont add all the files from git , add the Code files only , by mistake you might have done git add . . Please remove them from the commit

File: src/main/java/stepdefination/Calender_StepDefination.java (hunk 1 above)
Comment (gem-aditya-Saraswat) [reply src/main/java/stepdefination/Calender_StepDefination.java:170]: Done

File: src/main/java/stepdefination/Calender_StepDefination.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            logger.info("An exception occurred!", e);
+            GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED", STATUS.FAIL);
+        }
+    }
+
+    @Then("^User verify expected select range between (.+) and (.+)$")
+    public void dateRangeVerify(String startDate, String endDate){
+        try{
+            Calender_Utils calUtils = new Calender_Utils();
+            calUtils.verifyDataForRange(startDate, endDate);
```
[the remark this answers is not in the document]
Comment (gem-aditya-Saraswat) [reply src/main/java/stepdefination/Calender_StepDefination.java:181]: Done

File: target/classes/features/Ticket_Creation.feature [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    Then click on create ticket
+    Then Fill Details <Subject>, <Description>
+    Then Select ticket type as Incident
+    Then Select Department as <Department>
+    Then User uploads file from <filepath>
+    Then User Clicks on Submit Button
+    Then User fetch the Incident ID
+    Examples:
+      | Subject       | Description               | Department  | filepath                                                   |
+      |  Sample ticket| Sample Description text   | IT          | C:\Users\Aditya.Saraswat\Documents\Upload\sample.doc       |
```
[the remark this answers is not in the document]
Comment (gem-aditya-Saraswat) [reply target/classes/features/Ticket_Creation.feature:25]: Done

**Pull request closed without merging**